src/MockConnection.ts

```diff
@@ -0,0 +1,44 @@
+export class MockConnection {
+  public __knexUid = Math.trunc(Math.random() * 1e6);
+  public readonly fakeConnection = true;
+
+  public transactionStack: string[] = [];
+
+  public beginTransaction(cb: () => void) { cb(); }
+  public commitTransaction(cb: () => void) { cb(); }
+
+  /**
+   * Points the transaction stack to the given transaction.
+   * 
+   * If the given transaction ID is no on the stack, push it to the top.
+   * If the given transaction ID is already in the stack, remove all the transactions above it.
+   * If the given transaction ID is undefined, clears the stack.
+   */
+  public pointStackTo(txId: string | undefined) {
+    if (txId === undefined) {
+      this.transactionStack.splice(0);
+
+      return;
+    }
+
+    const index = this.transactionStack.indexOf(txId)
+
+    if (index >= 0) {
+      this.transactionStack.splice(index + 1);
+    } else {
+      this.transactionStack.push(txId);
+    }
+  }
+
+  /**
+   * Transaction ID managed by Knex.
+   */
+  get __knexTxId(): string | undefined {
+    return this.transactionStack.at(-1);
```

Comment:
Not supported by Node 12, you can change it to `this.transactionStack[this.transactionStack.length - 1]`

Reply:
Would you accept adding `ES2022.Array` to the TS libs?

The new `.at` method is not only much cleaner but it is also properly typed (`T | undefined` instead of `T`). Using the indexing signature can lead to unchecked undefined values.

Reply:
Ok, no problem, let's add it to the `tsconfig`

Reply:
Sadly `es2022.array` is only available on TS 4.9, I tried updating, but that breaks some other things. It seems `ts-node`/`ts-node-dev` will have to change some things to work with the latest TS.

I'll add a method to have this safely typed and use the index signature.

Reply:
Took me some time because I had to compile Node 12 from source as it is no longer distributed pre-compiled for NixOS

Reply:
I don't think that it worth it, if we want the `.at` only for type inference, and it available on the latest TS it not worth to invest in it.

Reply:
I ended up replacing it with a very simple stack class